HW#2 – Vasilii Shabanov

shabspb opened the pull request:
This weather widget is created using the SCSS preprocessor, BEM technology, flexbox and grid CSS, bootstrap-reboot and gulp.

shabspb committed:
decision homework-2-responsive

shabspb committed:
modified: grid/src/css/style.min.css

shabspb committed:
update modified:   grid/src/css/style.min.css

shabspb committed:
made grid adaptive

shabspb committed:
final solution hw-2-responsive, added support for IE11

nvkuznetsova commented:
There are a lot of duplicate files in your project. Let's simplify your project structure. Please, reorganize it like this
![image](https://user-images.githubusercontent.com/31770506/94857009-82db7700-0439-11eb-930b-854c60ff75f6.png)
1. In `hw-2-responsive` create folder `[YOUR_NAME]`
2. All folders with common files should be in folder  `[YOUR_NAME]` (fonts, images, common styles)
3. Little comment on `gulpfile`. I'm not experienced in gulp but try to create separate tasks to build `flexbox` and `grid` solution. 
You may need to add scripts to `package.json` file. Here are examples of scripts for React app 
![image](https://user-images.githubusercontent.com/31770506/94857819-c5518380-043a-11eb-9827-79b993ad8858.png)
But in case configuring common `gulpfile` takes more than a couple of hours for you just leave separate gulpfiles in `flexbox` and `grid` folders.
I'll take a look at your homework anyway but it would be great if you do these changes. I will be less confused in the files and this will simplify the codereview.
Thanks in advance! 😊

shabspb committed:
changing the structure of files, modified gulpfile.js

shabspb commented:
I have changed the structure of files and modified all necessary files (gulpfile.js and package.json). 
I hope, code review will be simple now =))
I am very grateful to you for your comment because I learned a lot while correcting my project.

nvkuznetsova commented on Vasilii_Shabanov/src/flexbox/index.html:
```diff
@@ -0,0 +1,69 @@
+                    <div class="widget-item__city">Brooklyn / 55°</div>
+                </section>
+                <section class="widget-item widget-item_yellow">
+                    <img src="icons/Cloud-sun-snow.png" alt="Cloud and sun and snow" class="widget-item__icon_margin">
```
I haven't noticed that in the previous homework but is `widget-item__icon_margin` an element class name or a modifier? If it is a modifier class name then you should add also an element class name here as the modifier CSS class should always be used paired with the class of its block or element. And let's use boolean naming for modifier then `widget-item__icon_hasmargin` If it is an element class name then please rename it to `widget-item__icon-margin` as you use `_` to separate a modifier name and this is a little bit confusing.

shabspb replied:
It is a modifier, i have changed the modifier's name and added element's class name.

nvkuznetsova commented on Vasilii_Shabanov/src/flexbox/index.html:
```diff
@@ -0,0 +1,69 @@
+                </section>
+                <section class="widget-item widget-item_red">
+                    <img src="icons/Cloud-wind-moon.png" alt="Cloud and wind and moon"
+                        class="widget-item__icon widget-item__icon_cloudwindmoon">
```
For this modifier `widget-item__icon_cloudwindmoon` let's use key-value naming: `widget-item__icon_theme-cloudwindmoon`.

shabspb replied:
Have done

nvkuznetsova commented on Vasilii_Shabanov/src/flexbox/index.html:
```diff
@@ -0,0 +1,69 @@
+                        <div class="widget-item__everyday">
+                            <div class="widget-item__wrapper_day">mon</div>
+                            <img src="icons/Cloud-rain-mini.png" alt="Cloud and rain" class="widget-item__wrapper_icon">
+                            <div class="widget-item__wrapper_temperature">22°</div>
```
Are these classes: `widget-item__wrapper_day`, `widget-item__wrapper_icon`, `widget-item__wrapper_temperature` modifiers or elements? Pease fix the naming. You can read more about BEM naming here https://ru.bem.info/methodology/quick-start/#%D0%B2%D0%B2%D0%B5%D0%B4%D0%B5%D0%BD%D0%B8%D0%B5

shabspb replied:
I have changed element's names to widget-item__day, widget-item__temperature. Also i have changed modifier to widget-item__icon_nomargin.

nvkuznetsova commented on Vasilii_Shabanov/src/grid/sass/blocks/_widget.scss:
```diff
@@ -0,0 +1,146 @@
+            height: 64px;
+
+            &:nth-child(1) {
+                -ms-grid-column: 1;
```
I got a lint warning here
![image](https://user-images.githubusercontent.com/31770506/95026234-a18b7900-0698-11eb-8e2e-e6eeabc4ea32.png)
Seems you should always use standard properties with vendor-prefixed properties. More information about this rule here https://github.com/CSSLint/csslint/wiki/Require-standard-property-with-vendor-prefix
Please also add standard properties everywhere where you use vendor-prefixed ones.

shabspb replied:
I have added necessary standard properties.

nvkuznetsova commented:
Thanks a lot for refactoring the file structure! It was much easier to find files I should pay attention to.
Nice work on responsive styles, btw! 👍

shabspb committed:
changed modifier's and element's names, changed widget.scss for ie11, made one index.html for two layouts

shabspb commented:
Thank you so much! I have made some changing in the file structure, i united both index.html files to the one index.html file and moved it to src folder, i hope you enjoy it))
